spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/observation/web/reactive/WebFluxObservationAutoConfigurationTests.java

```diff
@@ -114,6 +123,121 @@ void afterMaxUrisReachedFurtherUrisAreDeniedWhenUsingCustomObservationName(Captu
+			});
+	}
+
+	@Test
+	void whenActuatorObservationsDisabledObservationsShouldNotBeRecordedUsingCustomEndpointBasePath() {
+		this.contextRunner.withUserConfiguration(TestController.class, TestObservationRegistryConfiguration.class)
+			.withConfiguration(AutoConfigurations.of(InfoEndpointAutoConfiguration.class,
+					WebFluxAutoConfiguration.class, HttpHandlerAutoConfiguration.class, EndpointAutoConfiguration.class,
+					WebEndpointAutoConfiguration.class, WebFluxEndpointManagementContextConfiguration.class,
+					MetricsAutoConfiguration.class, ObservationAutoConfiguration.class))
+			.withPropertyValues("management.endpoints.web.exposure.include=info",
+					"management.observations.http.server.actuator.enabled=false",
+					"management.endpoints.web.base-path=/management")
+			.run((context) -> {
+				assertThat(context).hasBean("actuatorWebEndpointObservationPredicate");
+				TestObservationRegistry observationRegistry = getInitializedTestObservationRegistry(context, "/test0",
+						"/management/info");
+				TestObservationRegistryAssert.assertThat(observationRegistry)
+					.hasNumberOfObservationsWithNameEqualTo("http.server.requests", 1)
+					.hasAnObservationWithAKeyValue("http.url", "/test0");
+			});
+	}
+
+	@Test
+	void whenActuatorObservationsDisabledObservationsShouldNotBeRecordedUsingCustomWebfluxBasePath() {
+		this.contextRunner.withUserConfiguration(TestController.class, TestObservationRegistryConfiguration.class)
+			.withConfiguration(AutoConfigurations.of(InfoEndpointAutoConfiguration.class,
+					WebFluxAutoConfiguration.class, HttpHandlerAutoConfiguration.class, EndpointAutoConfiguration.class,
+					WebEndpointAutoConfiguration.class, WebFluxEndpointManagementContextConfiguration.class,
+					MetricsAutoConfiguration.class, ObservationAutoConfiguration.class))
+			.withPropertyValues("management.endpoints.web.exposure.include=info",
+					"management.observations.http.server.actuator.enabled=false", "spring.webflux.base-path=/test-path")
+			.run((context) -> {
+				assertThat(context).hasBean("actuatorWebEndpointObservationPredicate");
+				TestObservationRegistry observationRegistry = getInitializedTestObservationRegistry("/test-path",
+						context, "/test0", "/actuator/info");
+				TestObservationRegistryAssert.assertThat(observationRegistry)
+					.hasNumberOfObservationsWithNameEqualTo("http.server.requests", 1)
+					.hasAnObservationWithAKeyValue("http.url", "/test0");
+			});
+	}
+
+	@Test
+	void whenActuatorObservationsDisabledObservationsShouldNotBeRecordedUsingCustomWebfluxBasePathAndCustomEndpointBasePath() {
+		this.contextRunner.withUserConfiguration(TestController.class, TestObservationRegistryConfiguration.class)
+			.withConfiguration(AutoConfigurations.of(InfoEndpointAutoConfiguration.class,
+					WebFluxAutoConfiguration.class, HttpHandlerAutoConfiguration.class, EndpointAutoConfiguration.class,
+					WebEndpointAutoConfiguration.class, WebFluxEndpointManagementContextConfiguration.class,
```

Comment:
These two tests are failing. I'm suspicious it's because `WebTestClient` ignores the base path.